frontend/packages/dev-console/integration-tests/support/step-definitions/quck-starts/filter-quick-starts-catalog.ts

```diff
@@ -0,0 +1,93 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps';
+import { quickStartsPO } from '../../pageObjects';
+import { quickStartsPage } from '../../pages';
+
+When('user clicks on "View all quick starts" on Build with guided documentation card', () => {
+  quickStartsPage.quickStartsCatalog();
+});
+
+Then('user can see Quick Starts catalog page', () => {
+  cy.get(quickStartsPO.quickStartTitle).should('be.visible');
+});
+
+Then('user can see filter by keyword search bar', () => {
+  cy.get(quickStartsPO.filterKeyword).should('be.visible');
+});
+
+Then('user can see Status filter dropdown', () => {
+  cy.get(quickStartsPO.statusFilter).should('be.visible');
+});
+
+Given('user is at Quick Starts catalog page', () => {
+  quickStartsPage.quickStartsCatalog();
+});
+
+When('user clicks on filter by keyword search bar', () => {
+  cy.get(quickStartsPO.filterKeyword).click();
+});
+
+When('user enters {string}', (filterName: string) => {
+  quickStartsPage.filterByKeyword(filterName);
+});
+
+Then('user can see "Install the OpenShift Pipelines Operator" Quick Start is present', () => {
+  quickStartsPage.cardPresent(quickStartsPO.installOpenshiftPipelineOperatorCard);
+});
+
+Given('user is at Quick Starts catalog page', () => {
+  quickStartsPage.quickStartsCatalog();
+});
+
+When('user clicks on Status filter menu', () => {
+  quickStartsPage.status();
+});
+
+Then('user can see Complete, In progress and Not started categories', () => {
+  cy.get(quickStartsPO.statusDropdown).contains('Complete');
+  cy.get(quickStartsPO.statusDropdown).contains('In progress');
+  cy.get(quickStartsPO.statusDropdown).contains('Not started');
+});
+
+Given('user is at Quick Starts catalog page', () => {
+  quickStartsPage.quickStartsCatalog();
+});
+
+Given('user has completed "Getting started with a sample application" Quick Start', () => {
+  quickStartsPage.executeQuickStart(quickStartsPO.getStartedWithASampleApplicationCard);
+});
```

Comment:
If possible, a dynamic parameter would be cool here as well, or?

```suggestion
Given('user has completed {string} Quick Start', (quickStartName: string) => {
  quickStartsPage.executeQuickStart(quickStartsPO.getStartedWithASampleApplicationCard);
});
```

Reply:
Although we use static page objects, I've made it dynamic